gateway/src/apicast/gc.lua

```diff
@@ -0,0 +1,100 @@
+-- In LuaJIT and Lua 5.1, the __gc metamethod does not work in tables, it only
+-- works in "userdata". This module introduces a workaround to make it work
+-- with tables.
+
+local rawgetmetatable = debug.getmetatable
+local getmetatable = getmetatable
+local newproxy = newproxy
+local ipairs = ipairs
+local pairs = pairs
+local pcall = pcall
+local table = table
+local unpack = unpack
+local error = error
+
+local _M = {}
+
+local function original_table(proxy)
+  return rawgetmetatable(proxy).__table
+end
+
+local function __gc(proxy)
+  local t = original_table(proxy)
+  local mt = getmetatable(proxy)
+
+  if mt and mt.__gc then mt.__gc(t) end
+end
+
+local function __tostring(proxy)
+  local mt = getmetatable(proxy)
```

Comment:
Why not calling `tostring(original_table(proxy))` ?

Reply:
This was needed because I wasn't calling `setmetatable(t, metatable)` at the beginning of `set_metatable_gc`. I needed to do that, otherwise, __index would not work.
I fixed this in a new commit. Please check it. I'll squash it after review.